transport/socks5/stream_dialer_test.go

```diff
@@ -164,3 +166,67 @@ func testExchange(tb testing.TB, listener *net.TCPListener, destAddr string, req
 
 	running.Wait()
 }
+
+func TestConnectWithoutAuth(t *testing.T) {
+	// Create a SOCKS5 server
+	server := socks5.NewServer()
+
+	// Create SOCKS5 proxy on localhost port 8000
+	go func() {
+		err := server.ListenAndServe("tcp", "127.0.0.1:8000")
```

Comment:
Use port zero to let the system pick the port for you to avoid conflicts. 
You will need to get the port somehow. Usually the listener will give you that. Perhaps the server gives you that too. Otherwise you will need to create the listener first, then pass to the server.